File: server/datasets/java_source-code/Multiple-F1/organic_04_merge_sort.java
import java.util.Arrays;

public class Solution {
    // Standard recursive merge sort
public int[] mergeSort(int[] items_3) {
        if (items_3 == null || items_3.length <= 1) return items_3;
        int mid = items_3.length / 2;
        int[] left = mergeSort(Arrays.copyOfRange(items_3, 0, mid));
        int[] right = mergeSort(Arrays.copyOfRange(items_3, mid, items_3.length));
        
        int[] result = new int[items_3.length];
        int i = 0, j = 0, k = 0;
        while (i < left.length && j < right.length) {
            if (left[i] <= right[j]) {
                result[k++] = left[i++];
            } else {
                result[k++] = right[j++];
            }
        }
        while (i < left.length) result[k++] = left[i++];
        while (j < right.length) result[k++] = right[j++];
        return result;
    }
}

File: server/datasets/python_source-code/Multiple-F1/organic_04_merge_sort.py
# Standard recursive merge sort
def merge_sort(items_3):
    if len(items_3) <= 1:
        return items_3
    half_3 = len(items_3) // 2
    left = merge_sort(items_3[:half_3])
    right = merge_sort(items_3[half_3:])
    
    result = []
    i = j = 0
    while i < len(left) and j < len(right):
        if left[i] <= right[j]:
            result.append(left[i])
            i += 1
        else:
            result.append(right[j])
            j += 1
    result.extend(left[i:])
    result.extend(right[j:])
    return result
